crates/runtime/src/authorization.rs

```diff
@@ -0,0 +1,183 @@
+use crate::{RuntimeError, RuntimeResult};
+use holochain::prelude::InstalledAppId;
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use std::fs::File;
+use std::io::{Read, Write};
+use std::ops::Deref;
+use std::path::PathBuf;
+use std::sync::{Arc, RwLock};
+
+/// A unique identifier representing the client instance
+/// For example if the client is an android app, the client id would be the package name
+/// i.e. "org.holochain.androidserviceclient.app"
+#[derive(Eq, Hash, PartialEq, Clone, Serialize, Deserialize, Debug)]
+pub struct ClientId(pub String);
+
+/// A map of ClientIds authorized to make app api requests to the specified InstalledAppIds
+#[derive(Clone, Serialize, Deserialize, Debug)]
+struct AuthorizedAppClients(HashMap<ClientId, Vec<InstalledAppId>>);
+
+pub struct AuthorizedAppClientsManager {
+    authorized_app_clients: Arc<RwLock<AuthorizedAppClients>>,
+    persisted_path: PathBuf,
+}
+
+const PERSISTED_FILE_NAME: &str = "authorized_app_clients.msgpack";
```

Comment:
Is msgpack necessarily the best format for this? I question its choice in Holochain and if this is unrelated to Holochain then I'd be tempted to use a more common format like JSON

Reply:
Sure I can switch to json. I was just looking for the best performance serde serialize (and had trouble integrating postcard), but this is a tiny amount of data so I'm sure its not relevant.

Reply:
switched to json: https://github.com/holochain/android-service-runtime/pull/99/commits/c0b062ccc856a48e82a55bb2535cd7a62db7c7d2